production/kubernetes/build/templates/tempo/main.jsonnet

```diff
@@ -0,0 +1,38 @@
+local agent = import 'grafana-agent/v1/main.libsonnet';
+
+{
+  agent:
+    agent.new('grafana-agent-traces', 'default') +
+    agent.withConfigHash(false) +
+    agent.withImages({
+      agent: (import 'version.libsonnet'),
+    }) +
+    agent.withTempoConfig({
+      receivers: {
+        jaeger: {
+          protocols: { thrift_compact: null, grpc: null },
+        },
+      },
+    }) +
+    agent.withTempoPushConfig({
+      endpoint: '${TEMPO_ENDPOINT}',
+      basic_auth: {
+        username: '${TEMPO_USERNAME}',
+        password: '${TEMPO_PASSWORD}',
+      },
+      batch: {
+        timeout: '5s',
+        send_batch_size: 1000,
+      },
+      queue: {
+        retry_on_failure: false,
+      },
```

Comment:
I'm not 100% sure about if these defaults will work for most people.

Reply:
yes, agree i don't know either.
batch timeout should be relatively low to make sure that spans make it to the backend in a timely manner when volume is low.  1000 was chosen semi randomly.  It matches our match accepted batch size in tempo ops.  This config will probably go to prod.

`retry_on_failure` is weird.  It really should be false.   The otlp grpc exporter already retries certain GRPC error codes and I don't know the exact mechanics for when and how the queue processor retries.  Also, generally in tracing, the volume is so high that it's preferred to start ditching spans to load shed then queue up a billion spans and continue retrying.